src/3_branch.js

```diff
@@ -16,47 +16,54 @@ if (config.CORDOVA_BUILD) { var exec = require("cordova/exec"); } // jshint igno
 
 var default_branch;
 
-/***
- * @param {Error} err
- * @param {function(?Error,?=)=} callback
- */
-function wrapError(err, callback) {
-	if (callback) { return callback(err); }
-	throw err;
-}
-
-/***
- * @param {function(?)} func
- * @param {function(?Error,?=)=} callback
- */
-function wrapErrorFunc(func, callback) {
-	return function(err, data) {
-		if (err && callback) { callback(err); }
-		else if (err) { throw err; }
-		else if (func) { func(data); }
+function wrap(parameters, func) {
+	/**
+	 * Enum for what parameters are in a wrapped Branch method
+	 * @enum {number}
+	 */
+	var callback_params = {
+	  NO_CALLBACK: 0,
+	  CALLBACK_ERR: 1,
+	  CALLBACK_ERR_DATA: 2
 	};
-}
 
-/***
- * @param {function(?Error)=} callback
- * @returns {function(?Error)}
- */
-function wrapErrorCallback1(callback) {
-	return function(err) {
-		if (err && !callback) { throw err; }
-		if (callback) { callback(err); }
-	};
-}
+	/***
+	 * @param {Error} err
+	 * @param {function(?Error,?=)=} callback
+	 */
+	function wrapError(err, callback) {
+		if (callback) { return callback(err); }
+		throw err;
+	}
 
-/***
- * @param {function(?Error,?)=} callback
- * @returns {function(?Error,?=)}
- */
-function wrapErrorCallback2(callback) {
-	/*** @type {function(?Error,?=)} */
-	var r = function(err, data) {
-		if (err && !callback) { throw err; }
-		if (callback) { callback(err, data); }
+	var r = function() {
+		var self = this, args, callback,
+		lastArg = arguments[arguments.length - 1];
+		if (parameters === callback_params.NO_CALLBACK || typeof lastArg != "function") {
+			callback = function(err) {
+				if (err) { throw(err); }
+			};
+			args = Array.prototype.slice.call(arguments);
+		}
+		else {
+			args = Array.prototype.slice.call(arguments, 0, arguments.length - 1) || [];
+			callback = lastArg;
+		}
+		self._queue(function(next) {
+			if (!func.init && !self.initialized) { return callback(new Error(utils.message(utils.messages.nonInit))); }
```

Comment:
a cleaner way to do this might be to just pass it in as a third parameter on the wrap function?

also -- you can move this line down to after the `var done` and just call `done(new Error)`- t thoughts